### PR TITLE
fix: take into account only active suspensions

### PR DESCRIPTION
**Pull request by marcellorigotti**
# Pull Request

Closes: PRO-xxx

## Checklist

Please conduct a thorough self-review before opening the PR.

- [x] I am confident that the code works.
- [x] I have written sufficient tests.
- [x] I have written and tested required migrations.
- [x] I have updated documentation where appropriate.

## Summary

Changed the way we count for suspensions, we now keep into account if the suspension is still valid, based on the current block.

### CODE REVIEW

File: state-chain/runtime/src/lib.rs [hunk 1]
```diff
@@ -2147,7 +2147,8 @@ impl_runtime_apis! {
 		}
 
 		fn cf_suspended_validators() -> Vec<(Offence, u32)> {
-			pallet_cf_reputation::Suspensions::<Runtime>::iter().map(|(key, elem)| (key, elem.len() as u32)).collect()
+			let current_block = <frame_system::Pallet<Runtime>>::block_number();
+			pallet_cf_reputation::Suspensions::<Runtime>::iter().map(|(key, elem)| (key, elem.iter().filter(|(suspended_until, _)| *suspended_until < current_block).collect::<Vec<_>>().len() as u32)).collect()
```
Comment (dandanlen) [inline state-chain/runtime/src/lib.rs:2151]: I think it should be `>` (or `>=`)

Comment (dandanlen) [reply state-chain/runtime/src/lib.rs:2151]:
```suggestion
			pallet_cf_reputation::Suspensions::<Runtime>::iter().map(|(key, elem)| (key, elem.iter().filter(|(suspended_until, _)| *suspended_until > current_block).count() as u32)).collect()
```

**Comment by marcellorigotti** [comment]
Is there a better way to filter out duplicates without collecting to a set?
I tried to use https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.unique, but with no std features those method are disabled..

**Comment by msgmaxim** [comment]
> Is there a better way to filter out duplicates without collecting to a set? I tried to use https://docs.rs/itertools/0.9.0/itertools/trait.Itertools.html#method.unique, but with no std features those method are disabled..

Not really, and I believe even `unique` from itertools uses a hashset internally.

File: state-chain/runtime/src/lib.rs [hunk 2]
```diff
@@ -2147,7 +2147,25 @@ impl_runtime_apis! {
...
+				if key == pallet_cf_threshold_signature::PalletOffence::FailedKeygen.into() {
+					return (key, suspended_for_keygen);
+				}
+				(key, elem.iter().filter_map(|(suspended_until, validator)| {
+					if *suspended_until > current_block {
+						return Some(validator);
+					}
+					None
+				}).collect::<BTreeSet<_>>().len() as u32)
+			}).collect()
```
Comment (dandanlen) [inline state-chain/runtime/src/lib.rs:2168]: FYI there is a pallet method `Reputation::<T>::validators_suspended_for` that might make this a bit more readable.

(`pub fn validators_suspended_for(offences: &[T::Offence]) -> BTreeSet<T::ValidatorId>`)

**Pull request merged**